src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -168,7 +168,11 @@ private BoundStatement RewriteEnumeratorForEachStatement(BoundForEachStatement n
                 var builder = ArrayBuilder<BoundExpression>.GetInstance(getEnumeratorInfo.Arguments.Length);
                 builder.Add(receiver);
                 builder.AddRange(getEnumeratorInfo.Arguments, 1, getEnumeratorInfo.Arguments.Length - 1);
-                getEnumeratorInfo = getEnumeratorInfo with { Arguments = builder.ToImmutableAndFree() };
+                getEnumeratorInfo = new MethodArgumentInfo(
+                                            getEnumeratorInfo.Method,
+                                            builder.ToImmutableAndFree(),
+                                            defaultArguments: default,
```

Comment:
Shouldn't we pass `getEnumeratorInfo.DefaultArguments` here?

Reply:
> Shouldn't we pass ```getEnumeratorInfo.DefaultArguments``` here?

Values in ```getEnumeratorInfo.DefaultArguments``` are obsolete since it is a map for the original argument list and we mutated it. We could, of course, adjust the map as well, but it will be a throw away effort because the map is not used for lowering, the arguments are already in place.